### PR TITLE
Add scatter test for datetime units

### PR DESCRIPTION
**Pull request by seohyeon-lee-2025**
## PR summary
Closes issue #26864 
added 3 subplots with datetime array on the x axis, y axis, and both axes.

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by seohyeon-lee-2025** [comment]
I incorporated the issues you mentioned in my 2nd commit.

**Comment by seohyeon-lee-2025** [comment]
output

File: lib/matplotlib/tests/test_datetime.py [hunk 1]
```diff
@@ -276,11 +276,38 @@ def test_quiverkey(self):
...
         ax.quiverkey(...)
 
-    @pytest.mark.xfail(reason="Test for scatter not written yet")
     @mpl.style.context("default")
     def test_scatter(self):
-        fig, ax = plt.subplots()
-        ax.scatter(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        base = datetime.datetime(2005, 2, 1)
+        dates = [base + datetime.timedelta(hours=(2 * i)) for i in range(732)]
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:283]: We probably don't need quite so long an array; maybe something like 10 items is good enough.

File: lib/matplotlib/tests/test_datetime.py [hunk 2]
```diff
@@ -276,11 +276,38 @@ def test_quiverkey(self):
...
+        N = len(dates)
+        np.random.seed(19680801)
+        y = np.cumsum(np.random.randn(N))
+        fig, axs = plt.subplots(3, 1, layout='constrained', figsize=(6, 6))
+        lims = [(np.datetime64('2005-02'), np.datetime64('2005-04')),
+                (np.datetime64('2005-02-03'), np.datetime64('2005-02-15')),
+                (np.datetime64('2005-02-03 11:00'), np.datetime64('2005-02-04 13:20'))]
+        axs[0].set_title('Axes.scatter used with datetime on x axis')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:291]: We probably don't need the title here; people won't be looking at the image by itself and can figure this out from the test context.

File: lib/matplotlib/tests/test_datetime.py [hunk 3]
```diff
@@ -276,11 +276,38 @@ def test_quiverkey(self):
...
+        # datetime array on x axis
+        axs[0].scatter(dates, y)
+        # rotate labels so they don't look clogged
+        for label in axs[0].get_xticklabels():
+            label.set_rotation(40)
+            label.set_horizontalalignment('right')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:297]: With a shorter array, this might not be necessary; would have to see what it looks like then.

**Comment by seohyeon-lee-2025** [comment]
output2

File: lib/matplotlib/tests/test_datetime.py [hunk 4]
```diff
@@ -276,11 +276,30 @@ def test_quiverkey(self):
...
-        fig, ax = plt.subplots()
-        ax.scatter(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        base = datetime.datetime(2005, 2, 1)
+        dates = [base + datetime.timedelta(hours=(2 * i)) for i in range(10)]
+        N = len(dates)
+        np.random.seed(19680801)
+        y = np.cumsum(np.random.randn(N))
+        fig, axs = plt.subplots(3, 1, layout='constrained', figsize=(6, 6))
+        lims = [(np.datetime64('2005-02'), np.datetime64('2005-04')),
```
Comment (scottshambaugh) [inline lib/matplotlib/tests/test_datetime.py:288]: It doesn't look like lims is used

**Comment by scottshambaugh** [comment]
Thank you for the contribution! I noted one thing to clean up but it looks generally good to me. 

A note for whoever merges this, it closes the "Axes.scatter" item of https://github.com/matplotlib/matplotlib/issues/26864, not the whole ticket.

**Comment by seohyeon-lee-2025** [comment]
I deleted lims in my most recent pull request. Please review my request again.

**Comment by QuLogic** [comment]
Thanks @seohyeon-lee-2025! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**